controllers/component_controller.go

```diff
@@ -291,8 +298,137 @@ func (r *ComponentReconciler) Reconcile(ctx context.Context, req ctrl.Request) (
 		}
 	}
 
+	if component.Spec.Build.ContainerImage == "" {
+		component.Spec.Build.ContainerImage = "quay.io/redhat-appstudio/user-workload:" + component.Namespace + "-" + component.Name
+	}
+
+	workspaceStorage := commonStorage("appstudio", component.Namespace)
+	pvc := &corev1.PersistentVolumeClaim{}
+	err = r.Get(ctx, types.NamespacedName{Name: workspaceStorage.Name, Namespace: workspaceStorage.Namespace}, pvc)
+	if err != nil {
+		if errors.IsNotFound(err) {
+			err = r.Client.Create(ctx, &workspaceStorage)
+			if err != nil {
+				log.Error(err, fmt.Sprintf("Unable to create common storage %v", workspaceStorage))
+				return ctrl.Result{}, err
+			}
+		} else {
+			log.Error(err, fmt.Sprintf("Unable to get common storage %v", workspaceStorage))
+			return ctrl.Result{}, err
+		}
+	}
+
+	triggerTemplate, err := triggerTemplate(component)
+	if err != nil {
+		log.Error(err, "Unable to generate triggerTemplate ")
+		return ctrl.Result{}, err
+	}
+
+	err = controllerutil.SetOwnerReference(&component, triggerTemplate, r.Scheme)
+	if err != nil {
+		log.Error(err, fmt.Sprintf("Unable to set owner reference for %v", triggerTemplate))
+	}
+	err = r.Get(ctx, types.NamespacedName{Name: triggerTemplate.Name, Namespace: triggerTemplate.Namespace}, &triggersapi.TriggerTemplate{})
+	if err != nil {
+		if errors.IsNotFound(err) {
+			err = r.Client.Create(ctx, triggerTemplate)
+			if err != nil {
+				log.Error(err, fmt.Sprintf("Unable to create triggerTemplate %v", triggerTemplate))
+				return ctrl.Result{}, err
+			}
+		} else {
+			log.Error(err, fmt.Sprintf("Unable to get triggerTemplate %s", triggerTemplate.Name))
+			return ctrl.Result{}, err
+		}
+	}
+
+	eventListener := eventListener(component, *triggerTemplate)
+
+	err = controllerutil.SetOwnerReference(&component, &eventListener, r.Scheme)
+	if err != nil {
+		log.Error(err, fmt.Sprintf("Unable to set owner reference for %v", eventListener))
+		return ctrl.Result{}, err
+	}
+	err = r.Get(ctx, types.NamespacedName{Name: eventListener.Name, Namespace: eventListener.Namespace}, &triggersapi.EventListener{})
+	if err != nil {
+		if errors.IsNotFound(err) {
+			err = r.Client.Create(ctx, &eventListener)
+			if err != nil {
+				log.Error(err, fmt.Sprintf("Unable to create eventListener %v", eventListener))
+				return ctrl.Result{}, err
+			}
+		} else {
+			log.Error(err, fmt.Sprintf("Unable to get eventListener %v", eventListener))
+			return ctrl.Result{}, err
+		}
+	}
+
+	initialBuildSpec := determineBuildExecution(component, paramsForInitialBuild(component), "initialbuildpath")
```

Comment:
could you pls help me understand the purpose of this intial build, how is it different to what was already generated during triggerTemplate above?

Reply:
The initial build is where a PipelineRun is kicked off right after the Component is created.
The subsequent ones are from github webhook triggers. 

The shape of the PipelineRun remains the same though.